Build error on NetBSD due to dl

**Describe the bug**
While attempting to update the version of suitesparse in pkgsrc I get the following error:
```
[ 93%] Linking C shared library libgraphblas.so
ld: cannot find -ldl
--- GraphBLAS/libgraphblas.so.9.1.0 ---
*** [GraphBLAS/libgraphblas.so.9.1.0] Error code 1
```
This happens sometimes and is very common. So much that we have a way to bypass these types of errors. That being said, it would be nice if this error were fixed upstream. NetBSD doesn't have libdl.

**To Reproduce**
Try to build wip/suitesparse on NetBSD.

**Expected behavior**
Since NetBSD doesn't have libdl, it would be expected that cmake would not check for libdl on NetBSD. I'm not sure if this is true on other BSDs.

**Screenshots**
If applicable, add screenshots to help explain your problem.

**Desktop (please complete the following information):**
 - OS: NetBSD
 - compiler: gcc 10.5
 - Version: 7.7.0

--- a/GraphBLAS/CMakeLists.txt
+++ b/GraphBLAS/CMakeLists.txt
@@ -409,13 +409,13 @@ if ( NOT NO_LIBM )
 endif ( )
 
 # libdl
-if ( NOT WIN32 )
+if ( NOT "${CMAKE_DL_LIBS}" STREQUAL "" )
     if ( BUILD_SHARED_LIBS )
-        target_link_libraries ( GraphBLAS PRIVATE dl )
+        target_link_libraries ( GraphBLAS PRIVATE ${CMAKE_DL_LIBS} )
     endif ( )
     if ( BUILD_STATIC_LIBS )
-        list ( APPEND GRAPHBLAS_STATIC_LIBS "dl" )
-        target_link_libraries ( GraphBLAS_static PUBLIC dl )
+        list ( APPEND GRAPHBLAS_STATIC_LIBS ${CMAKE_DL_LIBS} )
+        target_link_libraries ( GraphBLAS_static PUBLIC ${CMAKE_DL_LIBS} )
     endif ( )
 endif ( )
 

--- a/Mongoose/Include/Mongoose_Logger.hpp
+++ b/Mongoose/Include/Mongoose_Logger.hpp
@@ -103,10 +103,10 @@ typedef enum TimingType
 class Logger
 {
 private:
-    /* MONGOOSE_API */ static int debugLevel;
-    /* MONGOOSE_API */ static bool timingOn;
-    /* MONGOOSE_API */ static double clocks[6];
-    /* MONGOOSE_API */ static float times[6];
+    MONGOOSE_API static int debugLevel;
+    MONGOOSE_API static bool timingOn;
+    MONGOOSE_API static double clocks[6];
+    MONGOOSE_API static float times[6];
 
 public:
     static inline void tic(TimingType timingType);

